src/main/java/com/criteo/vips/enums/VipsAccess.java

```diff
@@ -0,0 +1,48 @@
+/*
+  Copyright (c) 2020 Criteo
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+package com.criteo.vips.enums;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public enum VipsAccess implements IEnumValue {
+    // can read anywhere
+    Random(0),
+    // top-to-bottom reading only, but with a small buffer
+    Sequential(1),
+    SequentialUnbuffered(2),
+    Last(3);
+
+    private int value;
+    private static Map map = new HashMap<>();
+
+    VipsAccess(int i) {
+      value = i;
+    }
+
+    static {
+        for (VipsAccess e : VipsAccess.values()) {
+            map.put(e.value, e);
+        }
+    }
+
+    public static VipsAccess valueOf(int i) {
+        return (VipsAccess) map.get(i);
+    }
+
+    @Override
+    public int getValue() {
+      return value;
+    }
+}
```

Comment:
I would favor end of file trailing newlines.

Reply:
I will fix it